components/portal/messages/filters.js

```diff
@@ -68,6 +68,31 @@ define(['angular'], function(angular) {
         });
       };
     })
+    .filter('filterMessagesWithInvalidDates', function() {
+      return function(messages) {
+          var thePresent = moment.now();
+          return messages.filter(function(message) {
+            var goDate = moment(message.goLiveDate);
+            var stopDate = moment(message.expireDate);
+            // If neither date is populated, message is valid.
+            if (!goDate.isValid() && !stopDate.isValid()) {
```

Comment:
copying from https://github.com/uPortal-Project/uportal-app-framework/pull/641#discussion_r156768897

Moment comparisons return booleans so this can be flattened to

```js
var now = moment();
var goDate = moment(message.goLiveDate);
var stopDate = moment(message.expireDate);

// If both dates are defined, check between
if (goDate.isValid() && stopDate.isValid()) {
  return now.isBetween(goDate, stopDate)
}

// If goLive defined, check after
if (goDate.isValid()) {
  return now.isAfter(goDate);
}

// If expire defined, check before
if (stopDate.isValid()) {
  return now.isBefore(stopDate);
}

// Has neither date defined, so don't filter it
return true;
```

Reply:
I like the comments inline. They don't go far enough. It's not readily apparent to my sleep-deprived brain what this filter does without a close reading, especially outside of this Pull Request with its supporting pull request description. Needs some kind of comment that says "Filters away not-yet-gone-live and expired messages." or so.